src/native/corehost/hostpolicy/hostpolicy_context.cpp

```diff
@@ -290,6 +290,16 @@ int hostpolicy_context_t::initialize(hostpolicy_init_t &hostpolicy_init, const a
         }
     }
 
+    // We pass the loaded hostfxr path to the SDK can load it without relying on dlopen/LoadLibrary to find it.
+    if (host_mode == host_mode_t::muxer &&
+        (pal::strcmp(get_filename(application).c_str(), _X("dotnet.dll")) == 0))
```

Comment:
How bad would it be to put the logic in `hostfxr` (since it is is the one that does the transformation to running an SDK command and it 'knows' when we're running the SDK) such that it passes the property over to `hostpolicy` instead of doing the file name comparison here?

It is unlikely that someone uses `dotnet` as their app name and even if they do, the only effect is that they get an additional property, so I expect this is fine. But I do think it would be nice to not need the special-casing in here when we already have the detection in `hostfxr`.

Reply:
I agree - this should ideally go into the code paths in hostfxr which handle SDK invocation (there's other special code there, for example parsing `global.json`)